File: solutions/r/log-levels/1/log-levels.R
library(stringr)

message <- function(msg) {
  sub("\\[.*\\]:\\s*", "", msg) |> trimws()
}

log_level <- function(msg) {
  sub("\\[(.*)\\].*", "\\1", msg) |> 
  trimws() |>
  str_to_lower()
}

reformat <- function(msg) {
  sprintf("%s (%s)", message(msg), log_level(msg))
}

File: solutions/fsharp/log-levels/1/LogLevels.fs
module LogLevels

let message (logLine: string): string =
    let start = logLine.IndexOf(":")
    logLine.[start+1..].Trim()

let logLevel(logLine: string): string = 
    let endLog = logLine.IndexOf("]")
    logLine.[1..endLog-1].ToLower()

let reformat(logLine: string): string = 
    let m = message logLine
    let l = logLevel logLine
    sprintf "%s (%s)" m l
